solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -363,13 +610,138 @@ pub struct InitializeUser<'info> {
         space = USER_ACCOUNT_SIZE
     )]
     pub user: Account<'info, User>,
+    #[account(seeds = [&base.to_bytes()[..32], CONTENT_NODE_SEED_PREFIX, &replica_set[0].to_le_bytes()], bump = replica_set_bumps[0])]
+    pub cn1: Account<'info, ContentNode>,
+    #[account(seeds = [&base.to_bytes()[..32], CONTENT_NODE_SEED_PREFIX, &replica_set[1].to_le_bytes()], bump = replica_set_bumps[1])]
+    pub cn2: Account<'info, ContentNode>,
+    #[account(seeds = [&base.to_bytes()[..32], CONTENT_NODE_SEED_PREFIX, &replica_set[2].to_le_bytes()], bump = replica_set_bumps[2])]
+    pub cn3: Account<'info, ContentNode>,
     #[account(mut)]
     pub authority: Signer<'info>,
     #[account(mut)]
     pub payer: Signer<'info>,
     pub system_program: Program<'info, System>,
 }
 
+/// Instruction container to initialize a content node account, 
+/// must be invoked from an existing Audius `admin` account.
+#[derive(Accounts)]
+#[instruction(base: Pubkey, sp_id: u16)]
+pub struct CreateContentNode<'info> {
+    pub admin: Account<'info, AudiusAdmin>,
+    #[account(
+        init,
+        payer = payer,
+        seeds = [&base.to_bytes()[..32], CONTENT_NODE_SEED_PREFIX, sp_id.to_le_bytes().as_ref()],
+        bump,
+        space = CONTENT_NODE_ACCOUNT_SIZE
+    )]
+    pub content_node: Account<'info, ContentNode>,
+    #[account()]
+    pub authority: Signer<'info>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+/// Instruction container for creating a content node with 3 proposers
+#[derive(Accounts)]
+#[instruction(base: Pubkey, p1: ProposerSeedBump, p2: ProposerSeedBump, p3: ProposerSeedBump, sp_id: u16)]
+pub struct PublicCreateContentNode<'info> {
```

Comment:
Can we try combining into CreateOrUpdateContentNode w/ init_if_needed - https://docs.rs/anchor-lang/latest/anchor_lang/derive.Accounts.html

Reply:
yes!